src/Tasks/FileIO/HashHelper.cs

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Security.Cryptography;
+
+namespace Microsoft.Build.Tasks
+{
+    internal class HashHelper
+    {
+        private static Dictionary<string, Func<HashAlgorithm>> _supportedAlgorithms
+            = new Dictionary<string, Func<HashAlgorithm>>(StringComparer.OrdinalIgnoreCase)
+            {
+                ["SHA256"] = () => new SHA256Managed(),
```

Comment:
Although I doubt this will be instantiated a lot, I think I'd prefer to see these be some sort of Lazy so that we don't end up with N instances of these classes when you might only be using one.  `Lazy<T>` still allocates, perhaps a factory method with a `switch`?  Is there a benefit of having them as static and would it be better to just instantiate them when needed?